### DISCUSSION TITLE
Reproduce semantic commit type on updates

### DISCUSSION THREAD

**Discussion by sclorng** [renovatebot/renovate#13712] [Request Help] [answered]
I have the current config 

```
"packageRules": [
    {
      "matchDepTypes": [
        "component"
      ],
      "groupName": "component",
      "extends": [
        ":semanticCommitScope(component)"
      ]
    },
    {
      "matchDepTypes": [
        "component"
      ],
      "matchUpdateTypes": [
        "major"
      ],
      "extends": [
        ":semanticCommitTypeAll(feat)"
      ],
      "commitBody": "BREAKING CHANGE: update component(s) to major version"
    },
    {
      "matchDepTypes": [
        "component"
      ],
      "extends": [
        ":semanticCommitTypeAll(feat)"
      ]
    },
    {
      "matchDepTypes": [
        "component"
      ],
      "matchUpdateTypes": [
        "patch"
      ],
      "extends": [
        ":semanticCommitTypeAll(fix)"
      ]
    }
  ]
```

What I'm trying to do is grouping major component updates to a breaking change PR. And grouping minor/patch updates, the commit type should be `fix` if the group only contains patch update and `feat` if at least one updates is a minor updates.

With the given config, it works more or less.
Major are isolated and it works well for them.
Minor and patch are isolated but it seems that depending the last update type of the group, the commit type will change from fix to feat, and fix to feat.
For eg. the history when a new component with a patch update is release shows  `changed title from feat(component): update component to fix(component): update component` while the PR still contains the previous minor component update.
Then, it changed from fix to feat on an another run.

Should we stick to only `feat!` for major and `fix` for minor/patch ?

**Comment by rarkins** [comment, marked as answer]
If the semantic types are mixed within a branch then the result is non-deterministic. I think you'll find it's what the *first* update in the branch has.

**Reply by sclorng** [reply]
Yes, it seems it is the same as with semanticPrefixFixDepsChoreOthers preset.

**Reply by sclorng** [reply]
Should it be considered a bug ?

**Reply by rarkins** [reply]
No. If you combine multiple types in the same branch then the result is non-deterministic.

**Reply by rarkins** [reply]
Renovate doesn't "understand" semantic types, such as to know that fix is more important than chore, etc.

**Reply by sclorng** [reply]
> Renovate doesn't "understand" semantic types, such as to know that fix is more important than chore, etc.

Ok, I do understand for my use case, it will not be possible to group and have a precedence order of commit type.

Regarding semanticPrefixFixDepsChoreOthers, then maybe updates should be splitted. Actually the description  is misleading as it let believe that dependencies will be split in different branches.
